server/protocols/webadmin/webadmin-data/src/main/java/org/apache/james/webadmin/routes/SieveScriptRoutes.java

```diff
@@ -0,0 +1,187 @@
+        @ApiResponse(code = HttpStatus.BAD_REQUEST_400, message = "Empty script is not accepted"),
+        @ApiResponse(code = HttpStatus.NOT_FOUND_404, message = "User not exist")
+    })
+    @ApiImplicitParams({
+        @ApiImplicitParam(
+            name = USER_NAME,
+            required = true,
+            paramType = "path",
+            dataType = "String",
+            example = ROOT_PATH + "/userNameA/" + SCRIPTS + "/scriptName1",
+            value = "Username"),
+        @ApiImplicitParam(
+            name = SCRIPT_NAME,
+            required = true,
+            paramType = "path",
+            dataType = "String",
+            example = ROOT_PATH + "/userNameA/" + SCRIPTS + "/scriptName1",
+            value = "Script name"),
+        @ApiImplicitParam(
+            required = false,
+            paramType = "query parameter",
+            dataType = "Boolean",
+            defaultValue = "False",
+            example = "?activate=true",
+            value = "If present, automatically activating the script.")
+    })
+    public void defineAddActiveSieveScript(Service service) {
+        service.put(USER_SCRIPT_PATH, this::addActiveSieveScript);
+    }
+
+    private HaltException addActiveSieveScript(Request request, Response response) throws UsersRepositoryException, QuotaExceededException, StorageException, ScriptNotFoundException {
+        User user = extractUser(request);
+        ScriptName script = extractScriptName(request);
+        sieveRepository.putScript(user, script, extractSieveScriptFromRequest(request));
+        if (isActivated(request.queryParams(ACTIVATE_PARAMS))) {
+            sieveRepository.setActive(user, script);
+        }
+        return halt(HttpStatus.NO_CONTENT_204);
+    }
+
+    private User extractUser(Request request) throws UsersRepositoryException {
+        Optional<String> userName = Optional.ofNullable(request.params(USER_NAME));
+        if (!userName.isPresent() || Strings.isNullOrEmpty(userName.get().trim())) {
+            throw400withInvalidArgument("Invalid username");
+        }
+
+        if (!usersRepository.contains(userName.get())) {
+            throw404("User not exist");
+        }
+
+        return User.fromUsername(userName.get());
+    }
+
+    private ScriptName extractScriptName(Request request) {
+        return Optional.ofNullable(request.params(SCRIPT_NAME))
+            .map(String::trim)
+            .filter(scriptName -> !Strings.isNullOrEmpty(scriptName))
+            .map(ScriptName::new)
+            .orElseThrow(() -> throw400withInvalidArgument("Invalid Sieve script name"));
+    }
+
+    private ScriptContent extractSieveScriptFromRequest(Request request) {
+        return new ScriptContent(request.body());
+    }
+
+    private boolean isActivated(String activateParam) {
+        Optional<String> activated = Optional.ofNullable(activateParam);
+        if (!activated.isPresent()) {
+            return false;
```

Comment:
Generally calling `isPresent` on optional is a code smell.

could you try to avoid it?